lms/djangoapps/courseware/views.py

```diff
@@ -609,14 +612,32 @@ def jump_to(request, course_id, location):
     # choose the appropriate view (and provide the necessary args) based on the
     # args provided by the redirect.
     # Rely on index to do all error handling and access control.
+
+    # this block is taken from upstream url_helpers.py - and it does not guarantee to work if anything
+    # (course_key, chapter, section, position, final_target_id) is not specified. It should be enough for prototyping
     if chapter is None:
-        return redirect('courseware', course_id=course_key.to_deprecated_string())
+        redirect_url = reverse('courseware', args=(unicode(course_key), ))
     elif section is None:
-        return redirect('courseware_chapter', course_id=course_key.to_deprecated_string(), chapter=chapter)
+        redirect_url = reverse('courseware_chapter', args=(unicode(course_key), chapter))
     elif position is None:
-        return redirect('courseware_section', course_id=course_key.to_deprecated_string(), chapter=chapter, section=section)
+        redirect_url = reverse(
+            'courseware_section',
+            args=(unicode(course_key), chapter, section)
+        )
     else:
-        return redirect('courseware_position', course_id=course_key.to_deprecated_string(), chapter=chapter, section=section, position=position)
+        # Here we use the navigation_index from the position returned from
+        # path_to_location - we can only navigate to the topmost vertical at the
+        # moment
+
+        redirect_url = reverse(
+            'courseware_position',
+            args=(unicode(course_key), chapter, section, position)
+        )
+
+    if final_target_id:
+        redirect_url += "?activate={final_target_id}".format(final_target_id=final_target_id)
```

Comment:
@e-kolpakov Maybe open an OSPR ticket for this specific change - to inform of the general intention & ask the question about the URL scheme. Note that you can include a portion of the specs if it helps explaining, but don't link to the full group work v2 specs document for now.

Note that this isn't solutions work - given the confusions in the past that might be worth mentioning it in the ticket, so that it's clear for Sarina from the start.